objcgen/NameGenerator.cs

```diff
@@ -120,5 +120,15 @@ public static string GetMonoName (Type t)
 				throw new NotImplementedException ($"Converting type {t.Name} to a mono type name");
 			}
 		}
+
+		public static string GetObjCParamName (Type pt, List<ProcessedType> allTypes)
```

Comment:
Would this be better named as something like GetObjCParamType?  since Parameters have a ParameterType property and a Name property, calling using ParamName in the method name when you're actually returning the ParamaterType could be confusing.  Especially with my own PR adding the GetExtendedParameterName method that returns the actual name as opposed to the type